File: backend/governance/src/main/java/com/commonsengine/governance/infrastructure/persistence/GovernanceRepositories.java
package com.commonsengine.governance.infrastructure.persistence;

import com.commonsengine.governance.domain.Model.ProposalStatus;
import com.commonsengine.governance.infrastructure.persistence.GovernancePersistence.ProposalEntity;
import com.commonsengine.governance.infrastructure.persistence.GovernancePersistence.VoteEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

/**
 * 治理域 JPA Repository 接口集合（匹配 Kotlin GovernanceRepositories.kt）。
 */
public final class GovernanceRepositories {

    private GovernanceRepositories() {
    }

    public interface ProposalRepository extends JpaRepository<ProposalEntity, String> {
        List<ProposalEntity> findByStatusOrderByCreatedAtDesc(ProposalStatus status);

        List<ProposalEntity> findByProposerIdOrderByCreatedAtDesc(String proposerId);
    }

    public interface VoteRepository extends JpaRepository<VoteEntity, String> {
        List<VoteEntity> findByProposalId(String proposalId);

        List<VoteEntity> findByProposalIdAndVoterId(String proposalId, String voterId);
    }
}

File: backend/governance/src/main/kotlin/com/commonsengine/governance/infrastructure/persistence/GovernanceRepositories.kt
package com.commonsengine.governance.infrastructure.persistence

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ProposalRepository : JpaRepository<ProposalEntity, Long> {

    /** 按提案 ID 查询 */
    fun findByProposalId(proposalId: String): ProposalEntity?

    /** 按状态查询提案 */
    fun findByStatus(status: com.commonsengine.governance.domain.ProposalStatus): List<ProposalEntity>
}

@Repository
interface VoteRepository : JpaRepository<VoteEntity, Long> {

    /** 按提案 ID 查询所有投票 */
    fun findByProposalId(proposalId: String): List<VoteEntity>

    /** 检查是否已投票（一人一票） */
    fun existsByProposalIdAndVoterId(proposalId: String, voterId: String): Boolean
}
